app/services/mentions/create_all.rb

```diff
@@ -10,32 +10,55 @@ def self.call(*args)
 
     def call
       # Only works for comments right now.
+      mentioned_users = users_mentioned_in_notifiable_text_that_are_existing_and_not_its_owners
+
+      delete_mentions_removed_from_notifiable_text(mentioned_users)
+      create_mentions_for(mentioned_users)
+    end
+
+    private
+
+    def users_mentioned_in_notifiable_text_that_are_existing_and_not_its_owners
+      mentioned_usernames = extract_usernames_from_mentions_in_text
+
+      collect_existing_users(mentioned_usernames).
+        then do |existing_mentioned_users|
+          reject_notifiable_owner existing_mentioned_users
+        end
+    end
+
+    def collect_existing_users(usernames)
+      usernames.filter_map do |username|
```

Comment:
We could find users with one query `User.where(username: usernames)`, am I right?

Reply:
Yes, we can 👍 This will also implicitly takes care of the filtering.

Reply:
Oh, such a great idea!